### PR TITLE
Remove support for macOS/OSX.

### PR DESCRIPTION
**Pull request by clalancette**
We haven't had support for it for years, so there is no reason to keep it in the scripts.  If we ever do reinstate support, we can always pull this from the git history.

### CODE REVIEW

**Review by mjcarroll — APPROVED** [review]

**Review by nuclearsandwich — APPROVED** [review]

File: ros2_batch_job/__main__.py [hunk 1]
```diff
@@ -137,7 +131,7 @@ def main(sysargv=None):
         ]
     else:
         build_function = build_and_test_and_package
-        if sys.platform in ('darwin', 'win32'):
+        if sys.platform in ('win32'):
```
[the remark this answers is not in the document]
Comment (clalancette) [reply ros2_batch_job/__main__.py:134]: > does this work with a one-tuple? Doesn't python usually auto-unbox these?

Good question, but this does seem to work fine:

```
>>> if 'foo' in ('foo'):
...     print('hello')
... 
hello
>>> if 'bar' in ('foo'):
...     print('hello')
... 
>>> 
```

[the remark this answers is not in the document]
Comment (clalancette) [reply ros2_batch_job/__main__.py:134]: That said, the internet seems to agree that a tuple is declared by the existence of `,`, not parentheses.  Thus, I'm going to add in the extra comma just to be totally clear.

[the remark this answers is not in the document]
Comment (clalancette) [reply ros2_batch_job/__main__.py]: Fixed in 70381d1.

**Comment by clalancette** [comment]
Test CI:
* Linux [Build Status](http://ci.ros2.org/job/test_ci_linux/150/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/test_ci_linux-aarch64/62/)
* Windows [Build Status](http://ci.ros2.org/job/test_ci_windows/322/)

**Pull request merged**